include/cutlass/arch/mma_sm60.h

```diff
@@ -84,7 +84,7 @@ struct Mma<
 
 #else
     CUTLASS_PRAGMA_UNROLL
-    for (int i = 0; i < 2; ++i) {
```

Comment:
Given that `size_t` is normally unsigned and the integer literal `2` is normally signed, it seems like this change would actually introduce a build warning, rather than fix one.  Is `size_t` signed on your platform?

Reply:
No, `size_t` is effectively `unsigned long`. But the compiler doesn't appear to like `int` used as an array index. It doesn't complain about mixing `size_t` with integer literals.

Reply:
> But the compiler doesn't appear to like int used as an array index.

What warning does it issue?

Reply:
Unfortunately, that doesn't seem to help. Either the overload is too similar for the compiler's taste...
```
/tmp/cutlass-git/include/cutlass/gemm/warp/mma_complex_tensor_op_fast_f32.h(161): error: more than one operator "[]" matches these operands:
            function "cutlass::Array<T, N, true>::operator[](int) const [with T=cutlass::complex<float>, N=8]"
/tmp/cutlass-git/include/cutlass/array.h(396): here
            function "cutlass::Array<T, N, true>::operator[](cutlass::Array<T, N, true>::size_type) const [with T=cutlass::complex<float>, N=8]"
/tmp/cutlass-git/include/cutlass/array.h(401): here
            operand types are: const cutlass::Array<cutlass::complex<float>, 8, true> [ cutlass::layout::ColumnMajor::LongIndex ]
          detected during:
            [instantiation contexts mercifully elided]
```
or the conversion warnings still occur. I tried having the operators take `int` only, deleting the `size_t` ones. But the `[]` operators in `include/cutlass/array_subbyte.h` also appear to be involved, and they're implemented using `at()` which also takes a `size_t`... this quickly turns into a rabbit hole.

Reply:
@iskunk Thanks for trying this out!  I appreciate you taking the time.

We'll try out some possible fixes with the most recent Clang as host compiler, and NVCC as device compiler.  That is currently the only supported LLVM configuration.  If we can't get it to work, then unfortunately we won't be able to accept these particular index type changes.

Reply:
@iskunk FYI, I added a unit test specifically to exercise `cutlass::Array` in host code, but couldn't replicate these build errors with Clang 17.0.6 (as the host compiler).

We have two reasons _not_ to add `operator[]` overloads to `cutlass::Array`.

1. It's not consistent with C++ Standard Library array types such as `std::array` and `std::vector`.
2. It introduces the possibility of ambiguity in `operator[]` overload resolution.

Do you normally get build warnings when using `int` indices with `std::array` or `std::vector`?